docs/dev/topology-over-raft.md

```diff
@@ -76,17 +76,17 @@ such as `check_and_repair_cdc_streams`.
 If there is no work for the state machine, tablet load balancer is invoked to
 check if we need to rebalance. If so, it computes an incremental tablet migration
 plan, persists it by moving tablets into transitional states, and moves the state machine
```

Comment:
> Load balancer will recognize decommissioning nodes and will
move tablet replicas away from such nodes with highest priority.

> Topology changes have now an extra step called "tablet draining" which
calls the load balancer. The step will execute tablet migration track
as long as there are nodes which require draining. It will not do regular
load balancing.

It should, if the regular load balancing doesn't interfere. For example, if there is a bootstrap in one rack and a decommission in another, there's no conflict and both can proceed. It may be true if bootstrap+decommission happen in the same rack.

We can adjust it later, no need to hold the series for this.


> If load balancer is unable to find new tablet replicas, because RF
cannot be met or availability is at risk due to insufficient node
distribution in racks, it will throw an exception. Currently, topology
change will retry in a loop. We should make this error cause topology
change to be paused so that admin becomes aware of the problem and
issues an abort on the topology change. There is no infrastructure for
aborts yet, so this is not implemented.

We should try to see this in advance and fail decommission early.

Reply:
> We should try to see this in advance and fail decommission early.

Yes.